YoutubeExplode/Exceptions/VideoRequiresPurchaseException.cs

```diff
@@ -0,0 +1,16 @@
+﻿using System;
+
+namespace YoutubeExplode.Exceptions
+{
+    /// <summary>
+    /// Thrown when the video is a paid Youtube Red video since then we have no access to more video details
+    /// </summary>
+    public class VideoRequiresPurchaseException : Exception
+    {
+        /// <inheritdoc />
+        public VideoRequiresPurchaseException()
+            : base()
```

Comment:
Let's add a message so the users can understand what went wrong:
`base("The video is a paid Youtube Red video and cannot be processed")`